### PR TITLE
Rescue rai from mappings nightmare

### PR DESCRIPTION
**Pull request by EnnuiL**
- SystemToast.Type (unmapped) has been refactored to SystemToast.Id since Type is a more misleading name while Id more accurately states the intention as a sort of toast id that differentiates toasts from each other
- MapIcon still needs to be completely refactored into MapDecoration (or the vice-versa..?)
- accumulatedScrollX/Y has been decopypasted from Mojmap; it's now scrollDeltaX/Y, especially since there is no accumulation going on here (unlike cursorDeltaX/Y, although after some discussion, that name is alright)

### CODE REVIEW

File: src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java
Comment (OroArmor) [inline src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java:1]: Why were these deleted?

Comment (EnnuiL) [reply src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java:1]: > Why were these deleted?

Made obsolete by Minecraft itself (Minecraft has LlamaVariants, setVariant asks for LlamaVariants as well; same case for parrots)

Comment (OroArmor) [reply src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java:1]: Thats right. Anything left in the constants project?

Comment (EnnuiL) [reply src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java:1]: > Thats right. Anything left in the constants project?

Only the package-info.java, really

Comment (OroArmor) [reply src/constants/java/org/quiltmc/quiltmappings/constants/LlamaVariants.java:1]: Yikes. that needs to be fixed

**Comment by eristhea** [comment]
my knight in shining armour

File: mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping [hunk 1]
```diff
@@ -1,12 +1,17 @@
 CLASS net/minecraft/unmapped/C_elvupmqi net/minecraft/client/gui/screen/BackupPromptScreen
 	FIELD f_bbxlfnbn showEraseCacheCheckbox Z
-	FIELD f_khazxlgf subtitle Lnet/minecraft/unmapped/C_rdaqiwdt;
+	FIELD f_fewgdmzl runnable Ljava/lang/Runnable;
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping]: this runnable is triggered when the ESC key is pressed and has code that closes resources
I'd call it something like `onCanceled`
```suggestion
	FIELD f_fewgdmzl onCanceled Ljava/lang/Runnable;
```

File: mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+	FIELD f_khazxlgf description Lnet/minecraft/unmapped/C_rdaqiwdt;
 	FIELD f_mblzcmck id I
 	FIELD f_okzzeovr wrappedText Lnet/minecraft/unmapped/C_ogfbkizf;
 	FIELD f_vwuzyxro eraseCacheCheckbox Lnet/minecraft/unmapped/C_lvxxsthn;
 	FIELD f_wwwbtzmt callback Lnet/minecraft/unmapped/C_elvupmqi$C_agdgbivq;
 	METHOD <init> (Ljava/lang/Runnable;Lnet/minecraft/unmapped/C_elvupmqi$C_agdgbivq;Lnet/minecraft/unmapped/C_rdaqiwdt;Lnet/minecraft/unmapped/C_rdaqiwdt;Z)V
+		ARG 1 runnable
```
Comment (eristhea) [inline mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping]:
```suggestion
```
can be removed and automapped

File: mappings/net/minecraft/client/toast/SystemToast.mapping [hunk 3]
```diff
@@ -16,24 +31,52 @@ CLASS net/minecraft/unmapped/C_vksdgrxp net/minecraft/client/toast/SystemToast
...
 		ARG 0 client
+		ARG 1 id
 		ARG 2 title
 		ARG 3 description
+	METHOD m_yghdjgwj hide ()V
+	METHOD m_zfyvjxrz (Lnet/minecraft/unmapped/C_sedilmty;Lnet/minecraft/unmapped/C_eqfjupcc;J)Lnet/minecraft/unmapped/C_zqhkdxaf$C_sakiolwc;
+		ARG 1 graphics
 	METHOD m_zstagntz getTextAsList (Lnet/minecraft/unmapped/C_rdaqiwdt;)Lcom/google/common/collect/ImmutableList;
 		ARG 0 text
+	CLASS C_ozahoshp Id
```
Comment (eristhea) [inline mappings/net/minecraft/client/toast/SystemToast.mapping:71]: I still prefer the `Type` name here. your argument for the new name doesn't really make sense to me
> a sort of toast id that differentiates toasts from each other
`type` is also a word that implies differentiation? I may be dense but I don't really get the refactor

**Review by eristhea — CHANGES_REQUESTED** [review]
looks great!

File: mappings/net/minecraft/client/toast/SystemToast.mapping (hunk 3 above)
Comment (eristhea) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: to expand my position a little, usually `id` is something more internal while `type` is something more user-facing

File: mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping (hunk 1 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping]: ah fair, although unfortunately the convention favors me and my beloved cancelled, not your canceled :p

File: mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping (hunk 2 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping]: Hm, am not sure about how to feel about relying on automappings; I'll keep it but with the correct mapping

File: mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping (hunk 1 above)
Comment (eristhea) [reply mappings/net/minecraft/client/gui/screen/BackupPromptScreen.mapping]: oh lol I corrected myself from the proper spelling because I thought american was 1 l

File: mappings/net/minecraft/client/toast/SystemToast.mapping (hunk 3 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: The issue of Type is that it implies that it can be recycled between toasts of a certain type while in fact? It's more of an identifier for one specific toast; The SystemToast.Type name had made me mistakenly use a vanilla toast "type" for a modded toast when in fact it shouldn't have been used that way; This is why I kinda resist against `Type`;

I guess this refactor could be reverted though, it's just that yeah, the original name bothers me; it feels misleading

Comment (eristhea) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: oh girl i did the exact same thing as you in my mod is that not what i'm supposed to do

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: > oh girl i did the exact same thing as you in my mod is that not what i'm supposed to do

You are not! You are supposed to create your own `SystemToast.<Thing>` and use it for toasts; If you use a Vanilla one? If the Vanilla one and your toast attempts to be added at the same time? One's gonna overwrite the other since they share the same id! Hence why am proposing the Mojmap-inspired `SystemToast.Id` (moj's way more verbose though): because it is effectively the id of a toast, and it is the method which allows toasts to be differentiated when added

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: Could you add a commend to `Toast#getId` clarifying that toasts with the same id will override one another?

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: Sure! I can Javadoc something; not sure if at `Toast#getId`, but I can try something

File: mappings/net/minecraft/client/toast/ToastManager.mapping [hunk 4]
```diff
@@ -6,7 +6,7 @@ CLASS net/minecraft/unmapped/C_eqfjupcc net/minecraft/client/toast/ToastManager
 		ARG 1 client
 	METHOD m_glbfkboq getToast (Ljava/lang/Class;Ljava/lang/Object;)Lnet/minecraft/unmapped/C_zqhkdxaf;
 		ARG 1 toastClass
-		ARG 2 type
+		ARG 2 id
 	METHOD m_lnyjvhnt clear ()V
 	METHOD m_lsmszfie getNotificationDisplayTime ()D
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/toast/ToastManager.mapping:14]:
```suggestion
	METHOD m_lsmszfie getNotificationDisplaySeconds ()D
```
I think either this^ or `SystemToast.Id.displayDuration` -> `SystemToast.Id.displayMilis`

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/toast/ToastManager.mapping (hunk 4 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/ToastManager.mapping:14]: I actually like your latter suggestion! Lemme see if I can apply it

File: mappings/net/minecraft/client/toast/SystemToast.mapping (hunk 3 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: I opted for a Javadoc on SystemToast.Id itself

**Review by supersaiyansubtlety — APPROVED** [review]

**Review by eristhea — APPROVED** [review]

File: mappings/net/minecraft/client/toast/SystemToast.mapping (hunk 3 above)
Comment (supersaiyansubtlety) [reply mappings/net/minecraft/client/toast/SystemToast.mapping:71]: Hm, doesn't it apply to any object returned as an id from `Toast#getId`, including the default `ID` in `Toast`?

I think it would make more sense to put it there, with an `@see` pointing to `SystemToast.Id`.

File: mappings/net/minecraft/client/toast/Toast.mapping [hunk 5]
```diff
@@ -1,9 +1,10 @@
 CLASS net/minecraft/unmapped/C_zqhkdxaf net/minecraft/client/toast/Toast
-	FIELD f_msurgqxf TYPE Ljava/lang/Object;
+	FIELD f_msurgqxf ID Ljava/lang/Object;
```
Comment (supersaiyansubtlety) [inline mappings/net/minecraft/client/toast/Toast.mapping]:
```suggestion
	FIELD f_msurgqxf DEFAULT_ID Ljava/lang/Object;
```

**Review by supersaiyansubtlety — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/client/toast/Toast.mapping (hunk 5 above)
Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/Toast.mapping]: Mojmap actually suggests that this weird Id is actually the id that is used on toasts that don't have the same system that `SystemToast` has and only really has one id that is used! Such as `RecipeToast`; It calls it `NO_TOKEN`, but i don't think it's a good name?

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/Toast.mapping]: oh well, placeholder name for now..? there should be a better name though

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/client/toast/Toast.mapping]: Do non-`SystemToast` toasts not override each other (afaict they all use this default id)?

I thought they did based on how `ToastManager#getToast` works.

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/Toast.mapping]: So, apparently they avoid this by using different classes alongside the ids? I still have to look at this more closely (something that is now impossible thanks to 24w14a branch being broken :p)

Comment (supersaiyansubtlety) [reply mappings/net/minecraft/client/toast/Toast.mapping]: But then they still override toasts of the same class (because they all have the same id).

Maybe this should be `NO_ID` with javadoc
> The default identifier used by toast classes for which only one instance is displayed at a time.

And a javadoc at `getId` could be something like
> Toasts with the same class with different identifiers can co-exist, while toasts sharing the same class and identifier one will overwrite each other.

(mostly copied from your javadoc)

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/Toast.mapping]: Ah fair; this matches with Mojang's intention of `NO_TOKEN` perfectly while not being confusing at the same time; I'll investigate this tomorrow, since a broken Enigma complicates things, and yeah, if I have to workaround, I'd rather work around it tomorrow morning :p

(btw, `getId` is once again a silly target, I'd find an another more appropriate method; clarifying Javadocs, however, are the correct path for this)

Comment (EnnuiL) [reply mappings/net/minecraft/client/toast/Toast.mapping]: I applied the `NO_ID` suggestion; however, the Javadoc should be done on another tackle on unmapped Toast stuff; it needs to be thought about carefully, but it should be worth it

**Review by supersaiyansubtlety — APPROVED** [review]
